app/src/main/java/org/mozilla/tv/firefox/settings/SettingsFragment.kt

```diff
@@ -42,6 +45,28 @@ class SettingsFragment : Fragment() {
                 telemetryButton.isChecked = state
             }
         })
+        settingsViewModel.events.observe(viewLifecycleOwner, Observer {
+            it?.consume { event ->
+                when (event) {
+                    Action.SESSION_CLEARED -> {
+                        // This is necessary because of several complex interactions
+                        // - For initialization logic to be started, all sessions must already be removed
+                        // - URL is set to app home by default when a new session is created
+                        // - Activity#recreate is posted to the event loop, and executed concurrently
+                        //
+                        // The end result is that, in a naive solution, sessions are removed, URL is set to home, the
+                        // WebView is (eventually) destroyed, and then the new WebView instance has no BackForwardHistory.
+                        // This causes the back button to be disabled until two sites have been visited. After many
+                        // attempted solutions, clearing the session both before and after recreate was the most workable.
+                        //
+                        // This is Bad Code. If you find a way to remove it, please do so.
```

Comment:
IMO, this comment belongs in the `doNotPersist` block in `SessionRepo`.  Calling `Activity#recreate` when session is cleared seems reasonable to me, it's the `SessionRepo` that's doing the strange stuff that needs to be explained / apologized for.